modguard/show.py

```diff
@@ -0,0 +1,38 @@
+import yaml
+from modguard.core.boundary import BoundaryTrie
+from typing import Any, Dict
+
+# This type hint only works on more recent versions
+# result_dict: TypeAlias = dict[str, str | bool | 'result_dict']
+
+def boundary_trie_to_dict(boundary_trie: BoundaryTrie) -> Dict[str, Any]:
```

Comment:
Typing w/ Any here because the recursively nested types weren't passing earlier python versions